src/server.rs

```diff
@@ -157,6 +155,46 @@ impl Server {
         ServerBuilder::default().build()
     }
 
+    /// Adds node to address space.
+    ///
+    /// # Errors
+    ///
+    /// This fails when the node cannot be added.
+    pub fn add_node<T: Attributes + DataType>(&self, node: &Node<T>) -> Result<ua::NodeId> {
+        let mut out_node_id: ua::NodeId = ua::NodeId::null();
+        let mut context: *mut NodeContext = ptr::null_mut();
+        if node.context().is_some() {
+            // SAFETY: `__UA_Server_addNode` takes `nodeContext` as a mutable pointer, but nothing
+            // is modified in the function. Here we cast the non-mut reference to mut
+            // Maybe the `nodeContext` could be modified internally, outside of the addNode function
+            // but this has not been clarified yet.
+            context =
+                unsafe { ptr::from_ref(node.context().as_ref().unwrap_unchecked()).cast_mut() };
```

Comment:
See the above comment.
Nothing seems like the context needs to be passed as mutable, but it has to be a mutable pointer.
For now I would work around this, but this may be a bug. We'd need to clarify upstream.
My guess is, that nothing will be modified by the server, the only option for modifying is `UA_Server_setNodeContext()` or doing it manually (with the raw pointer). And with these two options you are doing it actively and outside the `add_node` function which means the non-mut reference parameter should be fine.

Is it okay to leave it like that?